x/programs/rust/README.md

```diff
@@ -25,6 +25,100 @@ Just a couple things to note. Serialization is minimal, yet there are certain as
 
 On the Go side, we unmarshal in the same order.
 
+## Storage
+
+A Program has 2 storage types.
+
+- `Memory` in WebAssembly is a linear buffer of unsigned bytes that can read and
+written to by both the guest or host. Because of the very minimal types
+supported by WebAssembly Memory is critical for communication between `Program`
+and host.
+
+- `State` is a host module which is imported into the `Program` instance, which
+exposes access to a database store. Unlike `Memory` state can be durable.
+
+### Memory
+
+In this example we share a byte slice between our host and a `Program`. To
+achieve this we can write our byte slice to the shared memory from he host using
+an exported functions called `alloc` and `memory` To do this we would first
+allocate  a chunk of continuous memory of the length desired.  `Alloc` will
+return the `offset` of that slice.
+
+
+```go
+runtime := // create new runtime and initialize
+memory := runtime.Memory()
+
+bytes := []byte("hello world")
+bytesLen := uint64(len(bytes))
+
+// create a memory allocation 
+offset, _ := memory.Alloc(bytesLen)
+
+// copy bytes into memory allocation
+offset, _ = memory.Write(offset, bytes)
```

Comment:
I'm a little confused by calling both variables `offset`. From what I can tell, the first `offset` is actually a pointer to the start of the allocation of memory for the byte array. The second `offset` is actually an offset, a length between the start and finish of the new allocation. If this is correct, should we name the first offset differently?

Reply:
they are the same write should probably not return the offset as it is passed directly.

Reply:
They are the same `Write` should not return the offset as it is passed directly. Will update.